controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -71,12 +71,19 @@ module.exports = async (req, res) => {
 
     if (req?.userId) {
       copyUser.is_following = isFollowingResult;
-      if (copyUser.only_received_dm_from_user_following)
-        copyUser.isSignedMessageEnabled = isFollowingResult;
-      else copyUser.isSignedMessageEnabled = true;
     }
 
-    copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && copyUser.isSignedMessageEnabled;
+    copyUser.isSignedMessageEnabled = true;
+    copyUser.isAnonMessageEnabled = false;
+
+    if (copyUser.allow_anon_dm) {
+      if (copyUser.only_received_dm_from_user_following) {
+        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && isFollowingResult;
+      } else {
+        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new logic for setting `isSignedMessageEnabled` and `isAnonMessageEnabled` seems to be more accurate than the previous one. However, there's a potential issue with the current implementation. If `allow_anon_dm` is false, `isAnonMessageEnabled` will remain false regardless of the value of `only_received_dm_from_user_following`. This might not be the intended behavior if we want to allow anonymous messages only from followers when `only_received_dm_from_user_following` is true. Consider revising the logic to account for this scenario.
```diff
-    copyUser.isAnonMessageEnabled = false;
-    if (copyUser.allow_anon_dm) {
-      if (copyUser.only_received_dm_from_user_following) {
-        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && isFollowingResult;
-      } else {
-        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
-      }
-    }
+    if (copyUser.only_received_dm_from_user_following) {
+      copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && isFollowingResult;
+    } else {
+      copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->